### PR TITLE
feat(optimization): generate library

### LINKED ISSUE #8752: Cloud Optimization API: `cloudoptimization.googleapis.com`

### PR DESCRIPTION
**Pull request by dbolduc**
Fixes #8752 

There is some weirdness in the quickstart. I tried to be helpful.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/9002)

### CODE REVIEW

File: google/cloud/optimization/quickstart/WORKSPACE.bazel [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+# Add the necessary Starlark functions to fetch google-cloud-cpp.
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+# Fetch the Google Cloud C++ libraries.
+# NOTE: Update this version and SHA256 as needed.
+http_archive(
+    name = "com_github_googleapis_google_cloud_cpp",
+    sha256 = "a7e51bfffb95a377094b2ae7e3b9f715a68ed931c48992c7273b2fae989c029c",
+    strip_prefix = "google-cloud-cpp-1.36.0",
+    url = "https://github.com/googleapis/google-cloud-cpp/archive/v1.36.0.tar.gz",
```
Comment (dbolduc) [inline google/cloud/optimization/quickstart/WORKSPACE.bazel:28]: I can update this if there are other changes requested.

File: google/cloud/optimization/quickstart/quickstart.cc [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  if (!resp) throw std::runtime_error(resp.status().message());
+  std::cout << resp->DebugString() << "\n";
+
+  return 0;
+} catch (std::exception const& ex) {
+  std::cerr << "Standard exception raised: " << ex.what() << "\n";
+  return 1;
+}
+
+/**
```
Comment (coryan) [inline google/cloud/optimization/quickstart/quickstart.cc:50]: I wonder if we could build a quickstart around this data:

```
gsutil ls gs://cloud-samples-data/optimization-ai
```

and

https://cloud.google.com/optimization/docs/reference/rpc/google.cloud.optimization.v1#batchoptimizetoursrequest

which after several indirections allows you to use a file from GCS?

**Review by coryan — COMMENTED** [review]
The quickstart looks very long.  I suggested a different source for the data.

File: google/cloud/optimization/quickstart/WORKSPACE.bazel (hunk 1 above)
Comment (dbolduc) [reply google/cloud/optimization/quickstart/WORKSPACE.bazel:28]: Didn't do this. There will be a release soon anyway.

File: google/cloud/optimization/README.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  // solves the model, and writes the solution to the destination GCS bucket.
+  std::string const source =
+      "gs://cloud-samples-data/optimization-ai/async_request_model.json";
+  auto const destination =
+      std::string{"gs://"} + argv[2] + "/optimization_quickstart_solution.json";
+
+  namespace gc = ::google::cloud;
+  namespace optimization = ::google::cloud::optimization;
+
+  auto options = gc::Options{}.set<gc::UserProjectOption>(argv[1]);
```
Comment (coryan) [inline google/cloud/optimization/README.md:65]: Is this needed for service accounts?  If not, maybe we can just use the environment variable?

**Review by coryan — APPROVED** [review]
Feel free to merge as-is, we can iterate and improve later.

**Pull request merged**